### PR TITLE
.github: make govuln-check generic

### PR DESCRIPTION
**Pull request by vivekpatani**
Signed-off-by: vivekpatani <9080894+vivekpatani@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/govuln.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
       fail-fast: false
       matrix:
         target:
-          - linux-amd64-govuln-check
+          - govuln-check
```
Comment (ahrtr) [inline .github/workflows/govuln.yaml:10]: If you look at some other examples, such as [static-analysis.yaml](https://github.com/etcd-io/etcd/blob/main/.github/workflows/static-analysis.yaml), there is no target at all. 

It isn't a big deal to to me. Keeping it as it is or fixing it, either way works for me.

Comment (serathius) [reply .github/workflows/govuln.yaml:10]: I would prefer to go with simplicity.

Comment (ahrtr) [reply .github/workflows/govuln.yaml:10]: @vivekpatani  Please remove the target so as to make it as simple as possible. So that we can merge this PR.

Comment (ahrtr) [reply .github/workflows/govuln.yaml:10]: @vivekpatani please resolve this comment so that we can merge this PR. It should be a simple change.

Comment (vivekpatani) [reply .github/workflows/govuln.yaml]: @serathius @ahrtr removed everything but `run: date`, can remove that too. Let me know your thoughts. Sorry for the delay.

**Review by ahrtr — APPROVED** [review]
LGTM

**Pull request merged**